internal/controller/streams.go

```diff
@@ -197,105 +164,225 @@ func (sm *streamManager) BootUpStreamAndReconnect(ctx context.Context, cancel co
 		return err
 	}
 	snapshotCompleted.Done()
+
+	<-ctx.Done()
+	return ctx.Err()
+}
+
+// StreamLogs handles the log stream.
+func (sm *streamManager) StreamLogs(ctx context.Context, cancel context.CancelFunc) error {
+	errCh := make(chan error, 1)
+
+	go func() {
+		errCh <- sm.bufferedGrpcSyncer.ListenToLogStream()
+	}()
+
+	select {
+	case <-ctx.Done():
+		cancel()
+		return ctx.Err()
+	case err := <-errCh:
+		if err != nil {
+			cancel()
+			return err
+		}
+	}
+	return nil
+}
+
+// StreamLogs handles the log stream.
+func (sm *streamManager) StreamCiliumNetworkFlows(ctx context.Context, cancel context.CancelFunc, ciliumNamespace string) error {
+	defer cancel()
```

Comment:
`cancel` is unnecessary in this method.
```suggestion
func (sm *streamManager) StreamCiliumNetworkFlows(ctx context.Context, ciliumNamespace string) error {
```